src/validation.cairo

```diff
@@ -67,9 +75,22 @@ impl TransactionValidatorImpl of TransactionValidator {
     }
 }
 
-fn block_hash(self: @ChainState, block: @Block, merkle_root: u256) -> Result<u256, ByteArray> {
-    // TODO: implement
-    Result::Ok(0)
+fn block_hash(
+    version: u32, previous_block: [u32; 8], merkle_root: [u32; 8], time: u32, bits: u32, nonce: u32
+    ) -> [
+    u32
+; 8] {
+    let mut header_data = ArrayTrait::<u32>::new();
+    header_data.append(version);
+    header_data.append(time);
```

Comment:
it should be:

1. version
2. previous_block
3. merkle_root
4. time
5. bits
6. nonce

Reply:
So previous_block and merkle_root is missing

Reply:
Use `append_span`, for example:
```
header_data.append_span(previous_block.span())
```